driver/node.go

```diff
@@ -130,6 +130,13 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 	if noFormat {
 		log.Info("skipping formatting the source device")
 	} else {
+		if d.validateAttachment {
+			isAttached, err := d.mounter.IsAttached(source)
+			if !isAttached || err != nil {
```

Comment:
Using a logical operator could be quite confusing to the user because the `err` variable in the output below can be nil.

Let's separate this out into a check for `err != nil` and another one for `!isAttached` with distinct (error/log) messages.

Reply:
Separated the conditions and added a distinct log message to improve  debuggability